src/software/ai/passing/pass_generator.h

```diff
@@ -7,135 +7,65 @@
-    // threading this class uses
-    PassGenerator& operator=(const PassGenerator&) = delete;
-    PassGenerator(const PassGenerator&)            = delete;
-
-    /**
-     * Create a PassGenerator with given parameters
-     *
-     * @param world The world we're passing int
-     * @param passer_point The point we're passing from
-     * @param pass_type The type of pass we would like to perform.
-     *                  NOTE: this will _try_ to generate a pass of the type given,
-     *                  but it is not guaranteed, and can change during pass
-     *                  execution because of Passer/Receiver decisions
-     * @param passing_config The dynamic parameter config to use to tune the PassGenerator
-     * @param run_deterministically If true, this disables all threading so that
-     *                              the same sequence of calls to this function always
-     *                              returns the same values, no matter how much time
-     *                              passes between calls
-     */
-    explicit PassGenerator(const World& world, const Point& passer_point,
-                           const PassType& pass_type,
-                           std::shared_ptr<const PassingConfig> passing_config,
-                           const bool run_deterministically = false);
+    static_assert(std::is_enum<ZoneEnum>::value,
+                  "PassGenerator: ZoneEnum must be an zone id enum");
 
+   public:
     /**
-     * Updates the world
+     * Creates a new PassGenerator with the given pitch_division.
      *
-     * @param world
-     */
-    void setWorld(World world);
-
-    /**
-     * Updates the point that we are passing from
+     * The PassGenerator will use this pitch division to guide initial random samples
+     * in each zone after the pitch has been divided.
      *
-     * @param passer_point the point we are passing from
+     * @param pitch_division The pitch division to use when looking for passes
      */
-    void setPasserPoint(Point passer_point);
+    explicit PassGenerator(
+        std::shared_ptr<const FieldPitchDivision<ZoneEnum>> pitch_division,
+        std::shared_ptr<const PassingConfig> passing_config);
 
     /**
-     * Set the id of the robot performing the pass.
+     * Creates a PassEvaluation given a world and a field pitch division.
      *
-     * This id will be used so we ignore the passer when determining where to
-     * pass to. We assume this robot is on the friendly team.
+     * NOTE: It is important that this function is able to run in 10ms.
      *
-     * @param robot_id The id of the robot performing the pass
-     */
-    void setPasserRobotId(unsigned int robot_id);
-
-    /**
-     * Set the target region that we would like to pass to
+     * This is because a pass is evaluated on the provided world and if the
+     * pass generator only completes an iteration of pass updates once every 5 seconds,
```

Comment:
why 5 seconds?

Reply:
this comment is just severely outdated, fixed it